parser/binary.go

```diff
@@ -13,54 +15,46 @@ type Placeholder struct {
 	Num         int  `json:"num" mapstructure:"num" msgpack:"num"`
 }
 
+func init() {
+	jsoniter.RegisterTypeEncoderFunc("types.BytesBuffer", func(ptr unsafe.Pointer, stream *jsoniter.Stream) {
+		bb := ((*types.BytesBuffer)(ptr))
+
+		bufList := stream.Attachment.([]types.BufferInterface)
+		_placeholder := &Placeholder{Placeholder: true, Num: len(bufList)}
+		stream.WriteVal(_placeholder)
+		stream.Attachment = append(bufList, bb)
+	}, nil)
+
+	jsoniter.RegisterTypeEncoderFunc("[]byte", func(ptr unsafe.Pointer, stream *jsoniter.Stream) {
```

Comment:
[]byte doesn't seem to be working as expected, maybe there's something wrong with my testing?

```golang
package main

import (
	"fmt"

	"github.com/zishang520/socket.io-go-parser/v2/parser"
)

func main() {
	en := parser.NewEncoder()

	id := uint64(1)
	Attachments := uint64(1)

	fmt.Println(en.Encode(&parser.Packet{
		Type:        parser.EVENT,
		Nsp:         "/a",
		Data:        []any{"any", []byte{1, 2, 3}},
		Id:          &id,
		Attachments: &Attachments,
	}))
}
```

need: [51-/a,1["any",{"_placeholder":true,"num":0}] ╔╗╚] 
output: [50-/a,1["any","AQID"]]

Reply:
kk fixed this up